### PR TITLE
Task as/j15 episode page add route

### PR DESCRIPTION
**Pull request by qwiz1**

### CODE REVIEW

File: backend/src/api/episodes/episodes.api.ts [hunk 1]
```diff
@@ -20,6 +20,22 @@ const initEpisodesApi = ({ apiRouter, episodeService }: Args): Router => {
     }),
   );
 
+  episodeRouter.get(
+    EpisodesApiPath.ID,
+    handleAsyncApi(async (req, res) => {
+      return res.json(await episodeService.getById(req.params.id)).status(HttpCode.OK);
```
Comment (Denidzio) [inline backend/src/api/episodes/episodes.api.ts]: It can be on one-line

**Review by Denidzio — DISMISSED** [review]

File: backend/src/api/episodes/episodes.api.ts (hunk 1 above)
Comment (qwiz1) [reply backend/src/api/episodes/episodes.api.ts]: Don't exactly understand what you mean

File: shared/src/common/enums/api/episodes-api-path.enum.ts [hunk 2]
```diff
@@ -1,5 +1,6 @@
 enum EpisodesApiPath {
   ROOT = '/',
+  ID = '/:id',
```
Comment (ghost) [inline shared/src/common/enums/api/episodes-api-path.enum.ts]:
```suggestion
  $ID = '/:id',
```
dynamic parametres should be with `$`

**Review by ghost — CHANGES_REQUESTED** [review]

File: backend/src/services/episode/episode.service.ts [hunk 3]
```diff
@@ -16,6 +16,10 @@ class Episode {
     return this.#episodeRepository.getAll();
   }
 
+  public getById(id: string): Promise<TEpisode> {
+    return this.#episodeRepository.getById(id);
```
Comment (ghost) [inline backend/src/services/episode/episode.service.ts]: if you don't find episode with the id you must to throw the http-error

**Review by ghost — CHANGES_REQUESTED** [review]

**Review by artem-totality — DISMISSED** [review]

File: backend/src/services/episode/episode.service.ts [hunk 4]
```diff
@@ -16,6 +17,14 @@ class Episode {
     return this.#episodeRepository.getAll();
   }
 
+  public async getById(id: string): Promise<TEpisode> {
+    const episode = await this.#episodeRepository.getById(id);
+    if (!episode) {
+      throw new HttpError({ status: 404, message: 'Episode does not exist!' });
```
Comment (ghost) [inline backend/src/services/episode/episode.service.ts]:
```suggestion
      throw new HttpError({ status: 404, message: 'Episode does not exist!' });
```
use http code for status
and create `ErrorMessage` enum in `backend/src/common/enums/app`

**Review by ghost — CHANGES_REQUESTED** [review]

File: backend/src/services/episode/episode.service.ts [hunk 5]
```diff
@@ -16,6 +19,14 @@ class Episode {
     return this.#episodeRepository.getAll();
   }
 
+  public async getById(id: string): Promise<TEpisode> {
+    const episode = await this.#episodeRepository.getById(id);
+    if (!episode) {
+      throw new HttpError({ status: HttpCode.NOT_FOUND, message: ErrorMessage.NOT_FOUND });
```
Comment (ghost) [inline backend/src/services/episode/episode.service.ts:25]: Previously you have the more readable error message. Let's replace it with `Entity not found.`

**Review by ghost — CHANGES_REQUESTED** [review]

**Review by ghost — APPROVED** [review]

**Review by yevhenprybytok — APPROVED** [review]

**Pull request merged**